scripts/dts/gen_defines.py

```diff
@@ -105,6 +106,63 @@ def parse_args():
     return parser.parse_args()
 
 
+def augment_node(node):
+    # Augment an EDT node with these zephyr-specific attributes, which
+    # are used to generate macros from it:
+    #
+    # - z_path_ident: a node identifer based on node.path, e.g.
+    #   a node with path /soc/foo@123 would get "PATH_SOC_FOO_123".
+    #
+    # - z_inst_idents: node identifiers based on the index of the node
+    #   within the EDT list of nodes for each compatible, e.g.:
+    #   ["INST_3_<NODE's_COMPAT>",
+    #    "INST_2_<NODE's_OTHER_COMPAT>"]
+    #
+    # - z_alias_idents: node identifiers based on any /aliases pointing to
+    #   the node in the devicetree source, e.g.:
+    #   ["DT_ALIAS_<NODE's_ALIAS_NAME>"]
+    #
+    # - z_legacy_ident: a legacy node identifier based on the node's
+    #   compatible, plus information from its unit address (or its
+    #   parent's unit address) or its name, and/or its bus. No example
+    #   given since this is a complex and questionably useful form of
+    #   generating an identifier.
```

Comment:
Wonder if these could be calculated as needed instead, e.g. with a function that takes a node and returns the path identifier. Maybe the `z_` could be dropped then too, if the functions are in `gen_defines.py`.

Reply:
I realize that efficiency is not necessarily a goal for this script, but one of the benefits of augmenting the tree is that you can reuse the results of `node.edt.compat2enabled[compat].index(node)` by stashing it in z_inst_idents instead of repeating it each time `node_instance_aliases()` is called.

Since your comments so far seem to be about specifics rather than the PR in general, I'll fold in your suggestions and keep working on it. When it's done, `node_instance_aliases()` will be removed entirely.